src/validator.py

```diff
@@ -0,0 +1,32 @@
+class Validator():
+    THREE_BY_THREE = "3x3"
+    EXIT = "exit"
+    MOVE = "move"
+    ERROR = "error"
+
+    def validate_move(self, move, board):
+        if board.type == self.THREE_BY_THREE:
+            return self.validate_3x3(move, board)
+
+    def validate_3x3(self, move, board):
+        if move.lower() == self.EXIT:
+            return [self.EXIT, None]
+        elif self.validate_3x3_input(move, board) and self.validate_3x3_move(move, board):
+             return [self.MOVE, move] 
+    
+        return [self.ERROR, move]
+
+    def validate_3x3_input(self, move, board):
+        valid_input = ["1", "2", "3", "4", "5", "6", "7", "8", "9"]
+        return move in valid_input  
+
+    def validate_3x3_move(self, move, board):
```

Comment:
These 3x3 concerns are what I'm referring to when I say feature envy. They know an awful lot about the board. Might it make sense to have a board specific validation class instead of a generic validation class that knows about any specific board type?

Reply:
Same thing goes for the 3x3 board presentation logic in your cli.py abstraction.

Reply:
Ah I see, was thinking having one Validator class to handle all validations was a good idea. But I can see how that can get really bloated really fast. What do you think about doing something similar to what I did with player and board, a builder class that determines what kind of validator we need and then have each validator board specific?

Reply:
That sounds like a great idea. Have a board specific validator and presenter.